docs/development/design/watch-api.md

```diff
@@ -0,0 +1,191 @@
+
+   { "type": "bookmark" }
+   ...
+   {
+     "type": "delete",
+     "value":  {
+       "user_id": "3833f886-fc40-44df-b4e3-644fa287ed2d",
+       "organization_id": "f13fe84d-9d6e-4dad-8464-bb30e9ef0194",
+       "public_key": "...",
+       "local_ip": "172.17.0.4:58664",
+       "tunnel_ip": "",
+       "child_prefix": null,
+       "relay": false,
+       "discovery": false,
+       "reflexive_ip4": "47.196.141.166",
+       "endpoint_local_address_ip4": "172.17.0.4",
+       "symmetric_nat": true,
+       "hostname": "device2",
+       "revision": 10247
+     }
+   }      
+   ...
+   ```
+
+In the above example, the client gets a bookmark event imediately indicating no changes had occured since revision 10245 but then a little while later a delete event is delivered.
+
+### Client Library Access
+
+The traditional non-watch API for listing devices in an organization is:
+
+```go
+devices, _, err := client.DevicesApi.
+   ListDevicesInOrganization(context.Background(), orgID).
+   Execute()
+```
+
+To use the `watch=true` version of the API, use:
+
+```go
+watch, _, err := client.DevicesApi.
+   ListDevicesInOrganization(context.Background(), orgID).
+   Watch()
+
+defer watch.Close()
+
+// get all the events..
+for {
+    kind, device, err := watch.Receive()    
+}
+
+```
+
+To simplify working against the event based *watch* operations we have implemented a Kubernetes style Informer API.
+
+A Informer in essence is a local cached copy of the resources that clients are interested in. The local cache will be refreshed via the watch operation. But the use of the watch API is occurring asynchronously and hidden from the clients.
+
+To use the Informer API
+
+```go
+informer := client.DevicesApi.
+   ListDevicesInOrganization(context.Background(), orgID).
+   Informer()
+
+devices, _, err := informer.Execute()
+```
+
+Notice that result of `informer.Execute()` is the same as the traditional list style api.  Additionally an informer provides a chanel via `informer.Changed()` that you can use to wait for when you can call `informer.Execute()` to get a changed device list.
+
+```go
+informer := client.DevicesApi.
+   ListDevicesInOrganization(ctx, orgID).
+   Informer()
+
+for {
+    select {
+        case <- ctx.Done():
+            return
+        case <- informer.Changed():
+            devices, _, err := informer.Execute()
+            ... process the devices ....
+    }
+}
+```
+
+### Apiserver Implementation of `watch=true`
+
+The HTTP request handler servicing the `ListDevicesInOrganization` will:
+
+1. Create a **SignalBus** (described later in this do) subscription to `/devices/org={orgId}`
+2. Select matching devices from the DB.  It will force device results to be ordered by `revision` so that the last selected result has the highest revision.
+3. Each result will be sent to the client as independent JSON document.  This will continue until it gets an empty result set from the database.
```

Comment:
```suggestion
3. Each result will be sent to the client as an independent JSON document.  This will continue until it gets an empty result set from the database.
```